### ISSUE TITLE
Renew PROD entrust certs

### ISSUE THREAD

**Issue by gormless87** [bcgov/nr-fom#421]
**Describe the task**
The FOM TLS certificates expire August 09, 2023 and must be renewed to prevent service disruption.

**Copied from FOM #70:  ??Step 1 not needed for 2023 renewal??**
1.	Follow these instructions in the certbot help page to remove the existing certbot install and replace with the version maintained by common services - https://github.com/BCDevOps/certbot
2.	Follow these instructions to find the billing codes, and submit a service desk ticket to get the entrust certs renewed for another year.  https://apps.nrs.gov.bc.ca/int/confluence/pages/viewpage.action?pageId=95654219

**Acceptance Criteria**
- [x] Certificates renewed
- [x] Update Entrust notification email address
- [x] Update renewal process documentation

**Additional context**
- NOTE: the FOM group mail will receive renewal notifications (likely in August, 2023) for two certs - only forward the non www renewal to the Service Desk.

**Comment by MCatherine1994** [comment]
The service desk ticket from the last time: https://apps.nrs.gov.bc.ca/int/jira/servicedesk/customer/portal/1/SD-42213, think we could start by submitting a new service desk ticket with all the information from the last ticket. The new expense authority for that acccount would be "Derek Lefler". And we need to confirm if the billing cost or information changed, maybe could get those information from "\\warehouse\IMTS\!Workgrp\iStore\Corporate Overhead, TCA Codes".

And the service desk should be able to figure these out and update our certs, the cerbot will detect the new certs automatically. We just need to change the configuration to change the notification email to us. Certbot config docs: https://github.com/BCDevOps/certbot/blob/6bcd2c70c696d2517c5f959abbaf09da61f78f05/docker/README.md#quick-start

**Comment by MCatherine1994** [comment]
New ticket created for renew the FOM prod certificate: https://apps.nrs.gov.bc.ca/int/jira/servicedesk/customer/portal/1/SD-77799

**Comment by MCatherine1994** [comment]
fyi....
![image.png](https://images.zenhubusercontent.com/6201a0f0c6815e7dfafa3f78/2269de29-f035-4643-a344-5c8b56967a29)

**Comment by MCatherine1994** [comment]
from Basil:

We looked into the certbot situation in backlog refinement. Findings:

1. certbot successfully renewed the FOM certs on Aug 2 fully automatically (no manual renewal required). So it is still working.
2. Every execution of certbot is failing due to openshift permission error when trying to clean up openshift resources (after performing the cert renewal check). Error from server (Forbidden): networkpolicies.networking.k8s.io is forbidden: User "system:serviceaccount:a4b31c-prod:certbot" cannot list resource "networkpolicies" in API group "networking.k8s.io" in the namespace "a4b31c-prod". New issue to be opened for this, Catherine will check with Derek for help.
3. the email configured as part of the certbot job has been changed to Catherine, but this doesn't seem to do anything - as Catherine wasn't notified. Kevin received a notification, so there must be a different configuration (perhaps part of the entrust setup). I suggested changing the emails to the FOM team distribution list.
4. Certbot did have a warning about the certficate renewal protocol (ACMEv1) being deprecated and eventually not being supported. I assume this corresponds to what Michelle mentioned above, although it looks like ACMEv1 is still working for FOM at least, for now. Warning from certbot: "Certbot is configured to use an ACMEv1 server (https://www.entrust.net/acme/api/v1/directory/98-YMBC-M0BH). ACMEv1 support is deprecated and will soon be removed. See https://community.letsencrypt.org/t/143839 for more information."

**Comment by MCatherine1994** [comment]
Got the help from Derek, added more permission to the certbot role to solve the permission error we got in certbot jobs for cleaning up. Tested with a manual job, runs successfully.

![image.png](https://images.zenhubusercontent.com/6201a0f0c6815e7dfafa3f78/90cbf941-d403-4e5d-847b-817c09c821a5)

**Comment by MCatherine1994** [comment]
I'm still not so sure how we could update the notification email. Entrust support team needs more information to help us, and I'm not so sure all about that. I asked our service desk as well to see if they can help with that.
1. Order Number or User ID/Name
2. Your organization or the organization name who purchased the product or service
3. The name of Entrust Product or Service you are requesting support for
4. Contact of who provided the Entrust Product or Service
5. URL and name of the service you are requesting support for

**Comment by MCatherine1994** [comment]
Renew certificate process:
- Create a service desk jira ticket with the following billing information, need to check with Michelle to confirm all these information are correct:
```
Would like to get entrust SSL certs for fom.nrs.gov.bc.ca renewed.

Billing Account details:
Client: 133
RC: 7124Z
Service Line: 30353
Project: FSA Fingerprint Team - 71RA322
Financial Reporting account: 177510
Expense Authority: Michelle Douville
```
- Wait the response from the service desk, and Michelle needs to approve for that certificate
- Once the certificate is approved in Entrust, wait the certbot in FOM openshfit prod environment to pick it up automatically, it runs every Sunday and Wednesday
- Verify the certbot pod log to confirm the certificate has been picked up and renewed successfully. The log screenshot for this time

![Screen Shot 2023-08-08 at 3.51.03 PM.png](https://images.zenhubusercontent.com/6201a0f0c6815e7dfafa3f78/dffc7d70-52b3-4970-b3a5-4d310bf1ba5f)
![image.png](https://images.zenhubusercontent.com/6201a0f0c6815e7dfafa3f78/1f468cda-8e60-4f56-b22d-ace9c45ec3fd)

**Comment by MCatherine1994** [comment]
As asked by the service desk team, created a new ticket for updating the notification email for FOM prod Entrust certification. 

https://apps.nrs.gov.bc.ca/int/jira/servicedesk/customer/portal/1/SD-82350
